ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -414,6 +414,7 @@ func (r *RayServiceReconciler) constructRayClusterForRayService(rayService *rayv
 		rayClusterAnnotations[k] = v
 	}
 	rayClusterAnnotations[common.EnableAgentServiceKey] = common.EnableAgentServiceTrue
+	rayClusterAnnotations[common.RayHAEnabledAnnotationKey] = "true"
```

Comment:
hmm this means any rayservice user need to provision an external redis cluster. can we only enable this when user put in this annotation? 
```
if rayServiceInstance.annotations.get(RayHAEnabledAnnotationKey) == "true"
    rayClusterAnnotations[common.RayHAEnabledAnnotationKey] = "true"
```

Reply:
this seems weird that we are explicitly setting this annotation to true.

If it is always true, then the annotation is useless. if it is user defined, it should not be hard-coded.

As for the naming convention, it is not typical to add the terms "key" or "annotation" in the annotation.

Please refer to [nginx annotations](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/) for some example annotations

Reply:
same question, do we need to enforce GCS HA? are we able to make this configurable by user?

Reply:
> naming convention

The actual name of the key as defined in `constant.go` is `ray.io/ha-enabled`.

> weird that we are explicitly setting this annotation to true

The thinking was probably that the service controller always sets the annotation to true on the RayClusters it manages -- so that it's valid to create a RayCluster without the annotation (but not with a RayService).

The issue is that then you have to set up the external Redis, which might be a little much here.

Reply:
I think it is good point. I will close this pr. Since right now, we already support user defined annotation.

Reply:
@brucez-anyscale not quite. The user defined annotation on RayService doesn't propagate to RayCluster. There is also no way to set that for RayCluster in rayClusterSpec's RayService spec.

Reply:
line 413 does propagate